lib/Dialect/TritonGPU/Transforms/OptimizeDotOperands.cpp

```diff
@@ -213,8 +213,14 @@ class UseShmemForScales
     }
 
     auto localLoad = getNextOp<triton::gpu::LocalLoadOp>(reshapeOp5D.getSrc());
-    if (!localLoad || !isTmemCopyCompatible(localLoad.getSrc().getType())) {
-      return failure();
+    if (localLoad) {
+      auto localAlloc = getNextOp<LocalAllocOp>(localLoad.getSrc());
+      bool usesTMAload = false;
+      if (localAlloc && localAlloc.getSrc())
+        usesTMAload =
+            getNextOp<DescriptorLoadOp>(localAlloc.getSrc()) != nullptr;
+      if (!isTmemCopyCompatible(localLoad.getSrc().getType(), usesTMAload))
+        return failure();
     }
     opOperand.assign(localLoad.getSrc());
     return success();
```

Comment:
You don't handle the `!localLoad` case any more?

Reply:
Good catch, thank you!